packages/common/src/store/cache/reducer.ts

```diff
@@ -130,59 +144,68 @@ export const mergeCustomizer = (
 }
 
 const actionsMap = {
-  [ADD_SUCCEEDED](
-    state: {
-      entries: { [x: string]: any }
-      uids: any
-      subscribers: any
-      idsToPrune: any
-    },
-    action: { entries: any[]; replace: any }
-  ) {
+  [SET_CACHE_TYPE](state: CacheState, action: SetCacheTypeAction) {
+    return {
+      ...state,
+      cacheType: action.cacheType
+    }
+  },
+  [ADD_SUCCEEDED](state: CacheState, action: AddSuccededAction) {
+    const { entries, replace } = action
+    const { cacheType } = state
     const newEntries = { ...state.entries }
     const newUids = { ...state.uids }
     const newSubscribers = { ...state.subscribers }
     const newIdsToPrune = new Set([...state.idsToPrune])
-
-    action.entries.forEach(
-      (e: {
-        id: string
-        metadata: { blocknumber: number }
-        uid: string | number
-      }) => {
-        // Don't add if block number is < existing
-        const existing = unwrapEntry(newEntries[e.id])
-        if (
-          existing &&
-          existing.blocknumber &&
-          e.metadata.blocknumber &&
-          existing.blocknumber > e.metadata.blocknumber
-        ) {
-          return
-        }
-
-        if (action.replace) newEntries[e.id] = wrapEntry(e.metadata)
-        else {
-          newEntries[e.id] = wrapEntry(
-            mergeWith(
-              {},
-              { ...unwrapEntry(state.entries[e.id]) },
-              e.metadata,
-              mergeCustomizer
-            )
-          )
-        }
-
-        newUids[e.uid] = e.id
-        if (e.id in newSubscribers) {
-          newSubscribers[e.id].add(e.uid)
+    const now = Date.now()
+
+    for (let i = 0; i < entries.length; i++) {
```

Comment:
What's the reason for changing this to a for loop?

Reply:
from what ive read its faster than a forEach, since we dont have to create function context for each one